app/src/org/commcare/android/database/SqlStorage.java

```diff
@@ -76,19 +69,26 @@ public SqlStorage(String table, Class<? extends T> ctype, DbHelper helper) {
     /* (non-Javadoc)
      * @see org.javarosa.core.services.storage.IStorageUtilityIndexed#getIDsForValue(java.lang.String, java.lang.Object)
      */
+    @Override
     public Vector<Integer> getIDsForValue(String fieldName, Object value) {
         return getIDsForValues(new String[] {fieldName}, new Object[] { value} );
     }
     
     public Vector<Integer> getIDsForValues(String[] fieldNames, Object[] values) {
+        SQLiteDatabase db;
+        try {
+            db = helper.getHandle();
+        } catch (SessionUnavailableException e) {
+            throw new UserStorageClosedException(e.getMessage());
+        }
```

Comment:
Handle the checked SessionUnavailableException and throw a runtime exception.
This pattern is seen a lot in this PR and effectively mimics the previous behavior of SessionUnavailableException when it used to be a runtime exception.

This pattern is used when it is unclear to me how we should handle the DB closing down.